pb/lmsysmd/dataset/v1/dataset.proto

```diff
@@ -0,0 +1,54 @@
+syntax = "proto3";
+
+package lmsysmd.dataset.v1;
+
+import "buf/validate/validate.proto";
+import "google/type/date.proto";
+
+service DatasetService {
+  rpc BatchCreateDataset(BatchCreateDatasetRequest) returns (BatchCreateDatasetResponse) {
+    // Duplicate datasets should not be created and id should not auto-increment
+    option idempotency_level = IDEMPOTENT; 
+  }
+}
+
+message BatchCreateDatasetRequest {
+  string google_sheet_url = 1 [ 
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.uri = true 
+  ];
+}
+
+message BatchCreateDatasetResponse {
+  repeated Dataset datasets = 1 [ (buf.validate.field).required = true ];
+}
+
+// Each dataset is grouped by model_id
+message Dataset {
+  uint32 id = 1 [ (buf.validate.field).required = true ];
+  ModelId model_id = 2 [ (buf.validate.field).required = true ];
+  string model_vendor = 3 [ (buf.validate.field).required = true ] ;
+  uint32 model_param_count_billion = 4 [ (buf.validate.field).required = true ];
+  google.type.Date model_release_date = 5 [ (buf.validate.field).required = true ];
+  map<string, string> meta = 6 [ (buf.validate.field).required = true];
```

Comment:
please see https://github.com/bufbuild/protovalidate/blob/main/proto/protovalidate/buf/validate/validate.proto

and add validations on max map length, max and min key and value lengths